netket/graph/grid.py

```diff
@@ -13,14 +13,44 @@
 # limitations under the License.
 
 import itertools
-from typing import List
+from dataclasses import dataclass
+from functools import partial
+from typing import Callable, List, Tuple
 
+from netket.utils.semigroup import Element, Identity, dispatch
```

Comment:
I'm vaguely trying to keep to the google coding style where you have a newline between types of imports (still, modules, local modules, local imports) 

```suggestion
from netket.utils.semigroup import Element, Identity, dispatch

```